File: commonUI/src/main/java/com/ionos/common_ui/utils/ContextUtils.java
package com.ionos.common_ui.utils;

import android.app.Activity;
import android.content.Context;
import android.content.ContextWrapper;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class ContextUtils {

	private ContextUtils() {
		throw new AssertionError();
	}

	@Nullable
	public static Activity getActivity(@NonNull Context context) {
		return getActivity(context, Activity.class);
	}

	@Nullable
	public static <T> T getActivity(@NonNull Context context, @NonNull Class<T> activityType) {
		if (activityType.isInstance(context)) {
			return activityType.cast(context);
		} else if (context instanceof ContextWrapper) {
			return getActivity(((ContextWrapper) context).getBaseContext(), activityType);
		} else {
			return null;
		}
	}

	public static boolean isActivityFinishing(Context context) {
		return context instanceof Activity && ((Activity) context).isFinishing();
	}
}

File: app/src/main/java/com/ionos/utils/context/ContextUtils.kt
@file:JvmName("ContextUtils")

package com.ionos.utils.context

import android.content.Context
import android.content.res.Configuration

fun Context.isDarkMode(): Boolean {
    val uiMode = resources.configuration.uiMode
    return (uiMode and Configuration.UI_MODE_NIGHT_MASK) == Configuration.UI_MODE_NIGHT_YES
}
